File: C++/sort-the-people.cpp
class Solution {
public:
    vector<string> sortPeople(vector<string>& names, vector<int>& heights) {
        map<int, string> people;
        int sz = names.size();
        for (int i = 0; i < sz; i++) {
            people[-heights[i]] = names[i];
        }
        vector<string> ans;
        for (auto it: people) {
            ans.push_back(it.second);
        }
        return ans;
    }
};

File: Go/sort-the-people.go
func sortPeople(names []string, heights []int) []string {
    people := make([]*person, len(names))
    for i := range names {
        people[i] = &person{names[i], heights[i]}
    }
    slices.SortFunc(people, func (a, b *person) int {
        return b.height - a.height
    })
    ans := make([]string, len(people))
    for i := range people {
        ans[i] = people[i].name
    }
    return ans
}

type person struct {
    name string
    height int
}